### PR TITLE
Feat/better results

### PR DESCRIPTION
**Pull request by acruikshank**
### Motivation

This is still a new repo and could use lots of improvements.

### What's in this PR

Lots of improvements. Specifically:

* Modified naming to reduce stutter.
* Cleaned imports
* Added `NodeMatching` method for times when you start with a node, but not a Cid.
* Fetch method now return immediately with separate results and errors channels.
* `NewFetcher` returns a value and Fetcher methods now take a value receiver
* `FetchResult` now provides the last block path and last block link, supporting Cid lists for traversals.

### CODE REVIEW

File: fetcher.go [hunk 1]
```diff
@@ -6,31 +6,34 @@ import (
...
+	"github.com/ipld/go-ipld-prime/traversal"
+	"github.com/ipld/go-ipld-prime/traversal/selector"
+	"github.com/ipld/go-ipld-prime/traversal/selector/builder"
 )
 
 // TODO: need to support sessions
 
 type Fetcher struct {
-	Exchange *bitswap.Bitswap
+	exchange *bitswap.Bitswap
```
Comment (hannahhoward) [inline fetcher.go:22]: hey I just realized -- this really should not be an Exchange. it should be a block service, no? Unless we want to have more specific control over local vs remote loading I think?

Comment (hannahhoward) [reply fetcher.go:22]: Actually I believe specifically this should be a blockservice.BlockGetter -- which covers both the session AND non session version

**Review by hannahhoward — COMMENTED** [review]
LGTM. I can't decide how I feel about removing the word fetch from the method names but I think I agree with the basic idea

File: fetcher.go (hunk 1 above)
Comment (acruikshank) [reply fetcher.go:22]: This will be addressed in #2 .

**Pull request merged**